src/storage/src/source/mysql/replication.rs

```diff
@@ -462,15 +470,168 @@ pub(crate) fn render<G: Scope<Timestamp = GtidPartition>>(
+                                            &config.config.connection_context.ssh_tunnel_manager,
+                                        )
+                                        .await?;
+                                    if let Some((err_table, err)) =
+                                        verify_schemas(&mut conn, &[(&table, table_desc)])
+                                            .await?
+                                            .drain(..)
+                                            .next()
+                                    {
+                                        assert_eq!(
+                                            err_table, &table,
+                                            "Unexpected table verification error"
+                                        );
+                                        trace!(%id, "timely-{worker_id} DDL change \
+                                                    verification error for {table:?}: {err:?}");
+                                        let gtid_cap = data_cap_set.delayed(new_gtid);
+                                        data_output
+                                            .give(
+                                                &gtid_cap,
+                                                ((*output_index, Err(err)), new_gtid.clone(), 1),
+                                            )
+                                            .await;
+                                        errored_tables.insert(table.clone());
+                                    }
+                                }
+                            // Detect `DROP TABLE [IF EXISTS] <tbl>, <tbl>` statements. Since
+                            // this can drop multiple tables we just check all tables we care about
+                            } else if first.eq_ignore_ascii_case("drop")
+                                && second.eq_ignore_ascii_case("table")
+                            {
+                                let mut conn = connection_config
+                                    .connect(
+                                        &format!("timely-{worker_id} MySQL "),
+                                        &config.config.connection_context.ssh_tunnel_manager,
+                                    )
+                                    .await?;
+                                for (dropped_table, err) in verify_schemas(
+                                    &mut conn,
+                                    &table_info
+                                        .iter()
+                                        .filter(|(t, _)| !errored_tables.contains(t))
+                                        .map(|(t, d)| (t, &d.1))
+                                        .collect::<Vec<_>>(),
+                                )
+                                .await?
+                                .drain(..)
+                                {
```

Comment:
It's almost never worth it (line wise) to inline a multi-line expression in the for loop declaration. I think giving names to the various parts will make this easier to follow:

```suggestion
                               let expected = table_info
                                        .iter()
                                        .filter(|(t, _)| !errored_tables.contains(t))
                                        .map(|(t, d)| (t, &d.1))
                                        .collect();
                                let schema_errors = verify_schemas(&mut conn, &expected).await?;
                                for (dropped_table, err) in schema_errors {
```